ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD.hs

```diff
@@ -339,6 +340,30 @@ forwardValues (UtxoValues values) (UtxoDiff diffs) =
       UedsIns       -> error "impossible! duplicate insert of key"
       UedsInsAndDel -> error "impossible! duplicate insert of key"
 
+-- | Transport a set of values by applying a valid difference as well as
+-- creating the values for keys that must be created in said differences.
+--
+-- Note that this fails via 'error' if the diff is invalid, eg it deletes a key
+-- that is not present in the argument or inserts a key that is already in the
+-- argument.
+forwardValuesAndKeys :: forall k v. (Ord k, HasCallStack) => UtxoValues k v -> UtxoKeys k v -> UtxoDiff k v -> UtxoValues k v
```

Comment:
Would it make sense to make a note of the coupling that exists between `RewoundKeySets`, `UnforwardedReadSets`, and this function?

Reply:
I think it should be enough with what I wrote now